code/modules/mining/aux_base_camera.dm

```diff
@@ -276,4 +276,277 @@ datum/action/innate/aux_base/install_turret/Activate()
+
+
+
+
+
+/obj/item/circuitboard/computer/base_construction/arena
+	name = "Arena Construction Board"
+	desc = "Allows the construction of hazards in a specially designated arena."
+
+
+
+/obj/machinery/computer/camera_advanced/arena_construction
+	name = "arena construction console"
+	desc = "Designed to maximize viewership. Ensure all participants have signed the omni-waiver."
+	networks = list("arena")
+	circuit = /obj/item/circuitboard/computer/base_construction/arena
+	off_action = new/datum/action/innate/camera_off/base_construction
+	jump_action = null
+	var/list/hazards = list()
+	var/datum/action/innate/arena_base/turfchange/lava/V = new
+	var/datum/action/innate/arena_base/turfchange/ice/I = new
+	var/datum/action/innate/arena_base/turfchange/slow/W = new
+	var/datum/action/innate/arena_base/turfchange/fast/A = new
+	var/datum/action/innate/arena_base/object/launcher/L = new
+	var/datum/action/innate/arena_base/object/slip/S = new
+	var/datum/action/innate/arena_base/object/freeze/Z = new
+	var/datum/action/innate/arena_base/object/damage/D = new
+	var/datum/action/innate/arena_base/object/fire/F = new
+	var/datum/action/innate/arena_base/object/snare/N = new
+	var/datum/action/innate/arena_base/object/emitter/E = new
+	var/datum/action/innate/clear_hazards/H = new
+
+
+//Aux base construction console
+/mob/camera/aiEye/remote/arena_construction
+	name = "arena holo-drone"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "construction_drone"
+	use_static = FALSE
+	var/area/starting_area
+
+/mob/camera/aiEye/remote/arena_construction/Initialize()
+	. = ..()
+	starting_area = get_area(loc)
+
+/mob/camera/aiEye/remote/arena_construction/setLoc(var/t)
+	var/area/curr_area = get_area(t)
+	if(curr_area == starting_area || istype(curr_area, /area/maintenance/arena))
+		return ..()
+
+/mob/camera/aiEye/remote/arena_construction/relaymove(mob/user, direct)
+	dir = direct //This camera eye is visible as a drone, and needs to keep the dir updated
+	..()
+
+/obj/machinery/computer/camera_advanced/arena_construction/CreateEye()
+	eyeobj = new /mob/camera/aiEye/remote/arena_construction(get_turf(src))
+	eyeobj.origin = src
+
+/obj/machinery/computer/camera_advanced/arena_construction/GrantActions(mob/living/user)
+	..()
+	if(V)
+		V.target = src
+		V.Grant(user)
+		actions += V
+	if(I)
+		I.target = src
+		I.Grant(user)
+		actions += I
+	if(W)
+		W.target = src
+		W.Grant(user)
+		actions += W
+	if(A)
+		A.target = src
+		A.Grant(user)
+		actions += A
+	if(L)
+		L.target = src
+		L.Grant(user)
+		actions += L
+	if(S)
+		S.target = src
+		S.Grant(user)
+		actions += S
+	if(Z)
+		Z.target = src
+		Z.Grant(user)
+		actions += Z
+	if(D)
+		D.target = src
+		D.Grant(user)
+		actions += D
+	if(F)
+		F.target = src
+		F.Grant(user)
+		actions += F
+	if(N)
+		N.target = src
+		N.Grant(user)
+		actions += N
+	if(E)
+		E.target = src
+		E.Grant(user)
+		actions += E
+	if(H)
+		H.target = src
+		H.Grant(user)
+		actions += H
```

Comment:
what the actual fuck can you roll all this into a loop

Reply:
Not without hindering maintainability

Reply:
???? start out with a list of types
for loop through the types and init them into a granted_actions list
then on granting the actions for loop and call .Grant(user)